packages/components/src/navigator/use-navigator.ts

```diff
@@ -13,11 +13,12 @@ import type { Navigator } from './types';
  * Retrieves a `navigator` instance.
  */
 function useNavigator(): Navigator {
-	const [ , setPath ] = useContext( NavigatorContext );
+	const { setNavigatorPath, setIsAnimating } = useContext( NavigatorContext );
 
 	return {
 		push( path, options ) {
-			setPath( { path, ...options } );
+			setIsAnimating( true );
+			setNavigatorPath( { path, ...options } );
```

Comment:
This approach seems to work better than using the `onAnimationStart` callback on the `motion.div` component:

- in my tests, the `onAnimationStart` callback would fire _after_ the animation starts, which caused the scrollbars to still appear from time to time (depending on the browser rendering loop)
- setting `isAnimating` to `true` in the `push` function ensures that the flag is applied _before_ the animation starts